### PR TITLE
Fix the logger that is imported for critical actions

### PR DESCRIPTION
**Pull request by CarlSchwan**
Currently it's just using the normal psr logger

Needed for https://github.com/nextcloud/groupfolders/pull/1917

### CODE REVIEW

File: lib/public/Files/Events/ACLChangedEvent.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (skjnldsv) [inline lib/public/Files/Events/ACLChangedEvent.php:1]: Copyright header ;)

File: apps/admin_audit/lib/Listener/GroupFolderActionListener.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (skjnldsv) [inline apps/admin_audit/lib/Listener/GroupFolderActionListener.php:1]: Copyright header ;)

File: lib/public/Files/Events/ACLChangedEvent.php (hunk 1 above)
Comment (PVince81) [reply lib/public/Files/Events/ACLChangedEvent.php:1]: I rather have the feeling that this got pushed by mistake and belongs to the other PR

Comment (CarlSchwan) [reply lib/public/Files/Events/ACLChangedEvent.php:1]: Yeah it does. Old code I need to remove

**Review by skjnldsv — APPROVED** [review]

File: apps/admin_audit/lib/AppInfo/Application.php [hunk 3]
```diff
@@ -79,6 +80,10 @@ public function __construct() {
 	}
 
 	public function register(IRegistrationContext $context): void {
+		$context->registerService(LoggerInterface::class, function (ContainerInterface $c) {
```
Comment (nickvergessen) [inline apps/admin_audit/lib/AppInfo/Application.php:83]: I dislike overload default services.

What I did in Activity with the IDBConnection was specifying my actual implementations, like this:
```suggestion
		$context->registerService(Action::class, function (ContainerInterface $c) {
			return new Action($c->get('AuditLogger'))
		}

		$context->registerService('AuditLogger', function (ContainerInterface $c) {
```

But works better in activity app as it was 2 classes to inject into. Here the action class has 10 classes extending it.

Comment (nickvergessen) [reply apps/admin_audit/lib/AppInfo/Application.php:83]: The current implementation e.g. breaks when you try to log something to the normal log in the background job etc.

Comment (CarlSchwan) [reply apps/admin_audit/lib/AppInfo/Application.php]: Just pushed a new implementation that instead uses a new interface `IAuditLogger`.This has the advantage of enforcing the type of logger used based on the context and moving the logger creation method instead in a new file.

**Comment by CarlSchwan** [comment]
Unit tests are failing but I'm not sure it's caused by this PR: `phpunit: command not found`

**Comment by nickvergessen** [comment]
> Unit tests are failing but I'm not sure it's caused by this PR: phpunit: command not found

The "Set up php 7.4" command errors:
> chmod: cannot access '/usr/local/bin/phpunit': No such file or directory
> ✗ phpunit Could not setup phpunit

I assume this is a temporary bug with actions?

But this here is not:

Bildschirmfoto von 2022-03-07 13-16-07

**Review by nickvergessen — APPROVED** [review]
As this fixes a bug, we need to backport it, right?

**Comment by CarlSchwan** [comment]
> As this fixes a bug, we need to backport it, right?

yes

**Comment by szaimen** [comment]
This PR might have broken 
image

**Comment by nickvergessen** [comment]
Add the import in https://github.com/nextcloud/server/pull/31454/commits/3e0c42e3b3bd3a55bd6a08c81335267df5b5ee74#diff-26f1e4805b227aa0936eaf6b5492ca8b166f852322243fb5d37b0d27e4c30f95R47 ?

**Pull request merged**